Update volume/nfs base image from centos:7 to centos:stream9

oyiz-michael opened the pull request:
## What type of PR is this?
/kind cleanup

## What this PR does / why we need it:
Updates the volume/nfs test image base from centos:7 to quay.io/centos/centos:stream9.

## Which issue(s) this PR fixes:
Fixes #131874

## Special notes for your reviewer

## Does this PR introduce a user-facing change?
```release-note
NONE

Additional documentation:
N/A - this is an infrastructure update to test images only.

oyiz-michael committed:
Update volume/nfs base image from centos:7 to centos:8

- Update BASEIMAGE from centos:7 to centos:8 for all architectures
- Bump VERSION from 1.4 to 1.5
- Addresses issue #131874 for updating outdated base images

CentOS 7 is EOL and centos:8 provides better security updates
and modern package versions for the NFS test image.

pacoxu commented:
https://endoflife.date/centos

IIUC, centos 8 is also end of life.

mkumatag commented:
> https://endoflife.date/centos
> 
> IIUC, centos 8 is also end of life.

+1, lets start using centos stream 9 - https://quay.io/repository/centos/centos?tab=tags

oyiz-michael committed:
Update to CentOS Stream 9 based on reviewer feedback

- Update BASEIMAGE to quay.io/centos/centos:stream9 for all architectures
- Update Dockerfile to use dnf (CentOS Stream 9 package manager)
- Update package name from /usr/bin/ps to procps-ng

Addresses reviewer feedback that CentOS 8 is also EOL.
CentOS Stream 9 provides ongoing updates and support.

mkumatag reviewed (approved):
/lgtm

oyiz-michael commented:
/ok-to-test

pacoxu commented:
/ok-to-test
/retest
